src/resolve/resolved-dns-dnssec.c

```diff
@@ -728,25 +817,22 @@ int dnssec_verify_rrset(
         /* OK, the RRs are now in canonical order. Let's calculate the digest */
         initialize_libgcrypt(false);
 
-        hash_size = gcry_md_get_algo_dlen(md_algorithm);
-        assert(hash_size > 0);
-
-        gcry_md_open(&md, md_algorithm, 0);
-        if (!md)
-                return -EIO;
+        f = open_memstream(&sig_data, &sig_size);
+        if (!f)
+                return -ENOMEM;
```

Comment:
hmm, given that this is somewhat performance-sensitive I figure it would be low-hanging but worthy to call __fsetlocking(f, FSETLOCKING_BYCALLER) on the memstream, to turn off all locking on it. For just concatenating a few bytes we really shouldn't pay the hefty locking price stdio makes us pay normally